pnr/openroad/meta.yaml

```diff
@@ -56,12 +56,13 @@ requirements:
     - fmt
     - libgomp                 [not osx]
     - llvm-openmp             [osx]
-    - qt                      [linux]
+    - qt >=5.0,<6.0
   run:
     # missing run_exports
     - {{ pin_compatible('libboost', min_pin='x.x', max_pin='x.x') }}
     - {{ pin_compatible('python', min_pin='x.x', max_pin='x.x') }}
     - spdlog >=1.9,<1.10
+    - qt >=5.0,<6.0
```

Comment:
That should be already done by the qt package https://github.com/AnacondaRecipes/qt-feedstock/blob/master/recipe/meta.yaml#L12

Reply:
When I installed it on my system, it wouldn't run unless I manually installed Qt:

```
❯ openroad -gui                                                                                                                                                      (eda)
dyld[62010]: Library not loaded: @rpath/libQt5Widgets.5.dylib
  Referenced from: <15F03BC3-AD43-34FB-9775-A95230997895> /opt/homebrew/Caskroom/miniconda/base/envs/eda/bin/openroad
  Reason: tried: '/opt/or-tools/lib64/libQt5Widgets.5.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/or-tools/lib64/libQt5Widgets.5.dylib' (no such file), '/opt/or-tools/lib/libQt5Widgets.5.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/or-tools/lib/libQt5Widgets.5.dylib' (no such file), '/opt/homebrew/Caskroom/miniconda/base/envs/eda/bin/../lib/libQt5Widgets.5.dylib' (no such file), '/opt/or-tools/lib64/libQt5Widgets.5.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/or-tools/lib64/libQt5Widgets.5.dylib' (no such file), '/opt/or-tools/lib/libQt5Widgets.5.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/or-tools/lib/libQt5Widgets.5.dylib' (no such file), '/opt/homebrew/Caskroom/miniconda/base/envs/eda/bin/../lib/libQt5Widgets.5.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS@rpath/libQt5Widgets.5.dylib' (no such file), '/opt/or-tools/lib64/libQt5Widgets.5.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/or-tools/lib64/libQt5Widgets.5.dylib' (no such file), '/opt/or-tools/lib/libQt5Widgets.5.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/or-tools/lib/libQt5Widgets.5.dylib' (no such file), '/opt/homebrew/Caskroom/miniconda/base/envs/eda/bin/../lib/libQt5Widgets.5.dylib' (no such file), '/opt/or-tools/lib64/libQt5Widgets.5.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/or-tools/lib64/libQt5Widgets.5.dylib' (no such file), '/opt/or-tools/lib/libQt5Widgets.5.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/or-tools/lib/libQt5Widgets.5.dylib' (no such file), '/opt/homebrew/Caskroom/miniconda/base/envs/eda/bin/../lib/libQt5Widgets.5.dylib' (no such file), '/usr/local/lib/libQt5Widgets.5.dylib' (no such file), '/usr/lib/libQt5Widgets.5.dylib' (no such file, not in dyld cache)
fish: Job 1, 'openroad -gui' terminated by signal SIGABRT (Abort)
```

Reply:
Can you share the installation logs (if you still have them)?

Reply:
```
sean in 🌐 Garlemald in conda-eda on  openroad-osx [$!?] via 🅒 eda
❯ conda install -c 'file:///opt/homebrew/Caskroom/miniconda/base/envs/eda/conda-bld' openroad                                                                        (eda)
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /opt/homebrew/Caskroom/miniconda/base/envs/eda

  added / updated specs:
    - openroad


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    openroad-2.0_6924_g037e4e3a4|                0        17.7 MB  file:///opt/homebrew/Caskroom/miniconda/base/envs/eda/conda-bld
    ------------------------------------------------------------
                                           Total:        17.7 MB

The following NEW packages will be INSTALLED:

  fmt                pkgs/main/osx-arm64::fmt-8.1.1-h525c30c_1
  libboost           pkgs/main/osx-arm64::libboost-1.73.0-h49e8a49_12
  llvm-openmp        pkgs/main/osx-arm64::llvm-openmp-14.0.6-hc6e5704_0
  openroad           conda-bld/osx-arm64::openroad-2.0_6924_g037e4e3a4-0
  spdlog             pkgs/main/osx-arm64::spdlog-1.9.2-h525c30c_0


Proceed ([y]/n)? y


Downloading and Extracting Packages

Preparing transaction: done
Verifying transaction: done
Executing transaction: done

sean in 🌐 Garlemald in conda-eda on  openroad-osx [$!?] via 🅒 eda  took 17s
❯ openroad -gui                                                                                                                                                      (eda)
dyld[62010]: Library not loaded: @rpath/libQt5Widgets.5.dylib
  Referenced from: <15F03BC3-AD43-34FB-9775-A95230997895> /opt/homebrew/Caskroom/miniconda/base/envs/eda/bin/openroad
  Reason: tried: '/opt/or-tools/lib64/libQt5Widgets.5.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/or-tools/lib64/libQt5Widgets.5.dylib' (no such file), '/opt/or-tools/lib/libQt5Widgets.5.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/or-tools/lib/libQt5Widgets.5.dylib' (no such file), '/opt/homebrew/Caskroom/miniconda/base/envs/eda/bin/../lib/libQt5Widgets.5.dylib' (no such file), '/opt/or-tools/lib64/libQt5Widgets.5.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/or-tools/lib64/libQt5Widgets.5.dylib' (no such file), '/opt/or-tools/lib/libQt5Widgets.5.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/or-tools/lib/libQt5Widgets.5.dylib' (no such file), '/opt/homebrew/Caskroom/miniconda/base/envs/eda/bin/../lib/libQt5Widgets.5.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS@rpath/libQt5Widgets.5.dylib' (no such file), '/opt/or-tools/lib64/libQt5Widgets.5.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/or-tools/lib64/libQt5Widgets.5.dylib' (no such file), '/opt/or-tools/lib/libQt5Widgets.5.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/or-tools/lib/libQt5Widgets.5.dylib' (no such file), '/opt/homebrew/Caskroom/miniconda/base/envs/eda/bin/../lib/libQt5Widgets.5.dylib' (no such file), '/opt/or-tools/lib64/libQt5Widgets.5.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/or-tools/lib64/libQt5Widgets.5.dylib' (no such file), '/opt/or-tools/lib/libQt5Widgets.5.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/or-tools/lib/libQt5Widgets.5.dylib' (no such file), '/opt/homebrew/Caskroom/miniconda/base/envs/eda/bin/../lib/libQt5Widgets.5.dylib' (no such file), '/usr/local/lib/libQt5Widgets.5.dylib' (no such file), '/usr/lib/libQt5Widgets.5.dylib' (no such file, not in dyld cache)
fish: Job 1, 'openroad -gui' terminated by signal SIGABRT (Abort)

sean in 🌐 Garlemald in conda-eda on  openroad-osx [$!?] via 🅒 eda
❯ conda install 'qt>=5.0,<6.0'                                                                                                                                       (eda)
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /opt/homebrew/Caskroom/miniconda/base/envs/eda

  added / updated specs:
    - qt[version='>=5.0,<6.0']


The following NEW packages will be INSTALLED:

  gettext            pkgs/main/osx-arm64::gettext-0.21.0-h826f4ad_0
  glib               pkgs/main/osx-arm64::glib-2.69.1-h514c7bf_2
  gst-plugins-base   pkgs/main/osx-arm64::gst-plugins-base-1.14.1-hf0a386a_0
  gstreamer          pkgs/main/osx-arm64::gstreamer-1.14.1-he09cfb7_0
  jpeg               pkgs/main/osx-arm64::jpeg-9e-h1a28f6b_0
  krb5               pkgs/main/osx-arm64::krb5-1.19.4-h8380606_0
  libclang           pkgs/main/osx-arm64::libclang-12.0.0-default_hc321e17_4
  libedit            pkgs/main/osx-arm64::libedit-3.1.20221030-h80987f9_0
  libllvm12          pkgs/main/osx-arm64::libllvm12-12.0.0-h12f7ac0_4
  libpng             pkgs/main/osx-arm64::libpng-1.6.37-hb8d0fd4_0
  libpq              pkgs/main/osx-arm64::libpq-12.9-h65cfe13_3
  nspr               pkgs/main/osx-arm64::nspr-4.33-hc377ac9_0
  nss                pkgs/main/osx-arm64::nss-3.74-h142855e_0
  pcre               pkgs/main/osx-arm64::pcre-8.45-hc377ac9_0
  qt                 pkgs/main/osx-arm64::qt-5.15.9-hca03da5_0
  qt-main            pkgs/main/osx-arm64::qt-main-5.15.2-ha2d02b5_7
  qt-webengine       pkgs/main/osx-arm64::qt-webengine-5.15.9-h2903aaf_4


Proceed ([y]/n)? y


Downloading and Extracting Packages

Preparing transaction: done
Verifying transaction: done
Executing transaction: done

sean in 🌐 Garlemald in conda-eda on  openroad-osx [$!?] via 🅒 eda  took 10s
✦ ❯ openroad -gui                                                                                                                                                    (eda)
qt.qpa.fonts: Populating font family aliases took 135 ms. Replace uses of missing font family "Monospace" with one that exists to avoid this cost.
OpenROAD 037e4e3a46ec91e8020445672d86afc9ff675cd6
This program is licensed under the BSD-3 license. See the LICENSE file for details.
Components of this program may be licensed under more restrictive licenses which must be honored.
^C⏎

Reply:
I removed this in the latest patch. Perhaps Conda local repositories just don't support dependencies this way.